aspnet/mvc/controllers/routing.rst

```diff
@@ -1,8 +1,738 @@
+
+Using the ``default`` route::
+
+   routes.MapRoute("default", "{controller=Home}/{action=Index}/{id?}");
+
+The route template:
+
+- ``{controller=Home}`` defines ``Home`` as the default ``controller``
+- ``{action=Index}`` defines ``Index`` as the default ``action``
+- ``{id?}`` defines ``id`` as optional
+
+Default and optional route parameters do not need to be present in the URL path for a match. See the  :doc:`Routing </fundamentals/routing>` for a detailed description of route template syntax.
+
+``"{controller=Home}/{action=Index}/{id?}"`` can match the URL path ``/`` and will produce the route values ``{ controller = Home, action = Index }``. The values for ``controller`` and ``action`` make use of the default values, ``id`` does not produce a value since there is no corresponding segment in the URL path. MVC would use these route values to select the ``HomeController`` and ``Index`` action::
+
+  public class HomeController : Controller
+  {
+    public IActionResult Index() { ... }
+  }
+
+Using this controller definition and route template, the ``HomeController.Index`` action would be executed for any of the following URL paths:
+
+- ``/Home/Index/17``
+- ``/Home/Index``
+- ``/Home``
+- ``/``
+
+The convenience method :dn:method:`~Microsoft.AspNetCore.Builder.MvcApplicationBuilderExtensions.UseMvcWithDefaultRoute`::
+
+  app.UseMvcWithDefaultRoute();
+
+Can be used to replace::
+
+  app.UseMvc(routes =>
+  {
+     routes.MapRoute("default", "{controller=Home}/{action=Index}/{id?}");
+  });
+
+``UseMvc`` and ``UseMvcWithDefaultRoute`` add an instance of :dn:cls:`~Microsoft.AspNetCore.Builder.RouterMiddleware` to the middleware pipeline. MVC doesn't interact directly with middleware, and uses routing to handle requests. MVC is connected to the routes through an instance of :dn:cls:`~Microsoft.AspNetCore.Mvc.Internal.MvcRouteHandler`. The code inside of ``UseMvc`` is similar to the following::
+
+   var routes = new RouteBuilder(app);
+
+   // Add connection to MVC, will be hooked up by calls to MapRoute.
+   routes.DefaultHandler = new MvcRouteHandler(...);
+
+   // Execute callback to register routes.
+   // routes.MapRoute("default", "{controller=Home}/{action=Index}/{id?}");
+
+   // Create route collection and add the middleware.
+   app.UseRouter(routes.Build());
+
+**[review: this section rewritten]**
+:dn:method:`~Microsoft.AspNetCore.Builder.MvcApplicationBuilderExtensions.UseMvc` does not directly define any routes, it adds a placeholder to the route collection for the ``attribute`` route. The overload ``UseMvc(Action<IRouteBuilder>)`` lets you add your own routes and also supports attribute routing. :dn:method:`~Microsoft.AspNetCore.Builder.MvcApplicationBuilderExtensions.UseMvcWithDefaultRoute` defines a default route and supports attribute routing. The :ref:`attribute-routing-ref-label` section includes more details on attribute routing.
+
+.. _routing-conventional-ref-label:
```

Comment:
This seems fine - I think we should add a sentence like 

`UseMvc` and all of its variations always add a placeholder for the attribute route - attribute routing is always available regardless of how you configure MVC

Feel free to reword, I want it to be clear that attribute routing always 'just works'